### DISCUSSION TITLE
打造漂亮的terminal環境 (Zsh + Oh My Zsh + powerlevel10k)

### DISCUSSION THREAD

**Discussion by gnsJhenJie** [1101-GS4719-C/Python-Community-GS4719-C#460] [General]
覺得作業系統內建的terminal很醜嗎?
<img width="500" alt="Screen Shot 2022-03-09 at 14 42 46" src="https://user-images.githubusercontent.com/32983347/157386850-a9d38bc5-e256-4c7c-9e81-44a5f1e32684.png">
想把terminal變得像下圖這樣好看的話下面教你怎麼做!
<img width="1000" alt="Screen Shot 2022-03-09 at 14 51 58" src="https://user-images.githubusercontent.com/32983347/157388068-dc2633d5-3538-4cde-9844-ed5c462a6354.png">

### `這個教學只適用Linux, macOS, Windows WSL... (Windows本身的cmd或powershell不適用ㄛ~)`
# 安裝步驟  (以Ubuntu 20.04示範)
## 更新套件
```shell
sudo apt update
sudo apt upgrade
# install required pacakges
sudo apt install vim curl git
```
## 安裝 Zsh
> macOS Catalina以後系統預設使用Zsh，因此可直接跳過這步
```shell
# install zsh
sudo apt install zsh
# 將zsh設為預設shell
chsh -s $(which zsh)
```
## 安裝 Oh My Zsh
[Oh My Zsh](https://github.com/ohmyzsh/ohmyzsh)是一個管理zsh的工具
```shell
wget https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh
sh install.sh
```
## 安裝powerlevel10k
[powerlevel10k](https://github.com/romkatv/powerlevel10k#oh-my-zsh)是一個zsh的主題包，也可以自己找其他主題喔~ 
```shell
git clone --depth=1 https://github.com/romkatv/powerlevel10k.git ${ZSH_CUSTOM:-$HOME/.oh-my-zsh/custom}/themes/powerlevel10k
```
接著，透過文字編輯器(可用vim, vi, nano...)打開`~/.zshrc`並設定`ZSH_THEME="powerlevel10k/powerlevel10k"`
## 設定powerlevel10k
```shell
zsh
```
下完指令後會自動跳出主題設定頁面(如果沒有出現設定頁面可以改下`p10k configure`)，接著依照畫面設定自己想要的畫面即可~~
<img width="787" alt="Screen Shot 2022-03-10 at 18 43 16" src="https://user-images.githubusercontent.com/32983347/157645479-4edd3fa8-a1a2-4507-9c0a-8a5314a98414.png">

**Comment by chia-kai-chang** [comment]
感謝，常常看到別人的MAC有漂亮的終端機，但是不知道怎麼實作。我來照步驟做做看~

**Reply by gnsJhenJie** [reply]
因為是用Ubuntu的指令示範的，所以apt這個指令用到macOS時，要改用[brew](https://brew.sh/)或是其他套件管理工具，其他步驟應該大致上都一樣~

**Reply by chia-kai-chang** [reply]
感恩~ 我來試試看~ 😂

**Comment by ThomasMeng0515** [comment]
windows系統的終端機也可以美化喔~ 參考下方連結:
https://ithelp.ithome.com.tw/articles/10230798

**Comment by ghost** [comment]
感覺之後可以實際套用看看效果如何，畢竟一成不變的視覺效果真的會影響打code效率XDD

**Comment by xkllkx** [comment]
謝謝分享，照著樓主分享的步驟做，真的成功了~

**Comment by Sazmafuji** [comment]
太神了，感謝分享，現在terminal 變超好看

**Comment by jason79461385** [comment]
用完後terminal變得花花綠綠的突然有點不習慣了 哈哈哈哈

**Comment by treeann** [comment]
漂亮的terminal連結超重要ㄝ

**Comment by liangyinglly** [comment]
用完後我怎麼覺得有點眼花撩亂哈哈哈 需要一點時間適應

**Comment by etin2468** [comment]
字樣可以改成自己覺得好看的樣子真的棒，而且選擇很多
